api/v1alpha1/thanosruler_types.go

```diff
@@ -77,6 +77,12 @@ type ThanosRulerSpec struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default:={"serviceMonitor":{"enable":true}}
 	FeatureGates *FeatureGates `json:"featureGates,omitempty"`
+	// PrometheusRuleSelector is the label selector to discover PrometheusRule CRDs.
+	// If not specified, no PrometheusRules will be loaded.
+	// Once detected, these rules are made into configmaps and added to the Ruler.
+	// +kubebuilder:default:={matchLabels:{"operator.thanos.io/prometheus-rule": "true"}}
+	// +kubebuilder:validation:Required
+	PrometheusRuleSelector metav1.LabelSelector `json:"prometheusRuleSelector,omitempty"`
```

Comment:
somethings here seem conflicting

1. If not specified, no PrometheusRules will be loaded.
2. +kubebuilder:validation:Required

I think it should be made optional and turned into a pointer?

Reply:
Right that line shouldn't be there.
The behavior for this, is that will always look for prometheusrule objects and is set to our label by default
But you can customize this label (which is something we would need, as it might be hard for users to update all their existing prometheusrule objects with a specific label.

Reply:
Added a rule to validate minimum of one label